doc/userguide/rules/rule-types.rst

```diff
@@ -0,0 +1,930 @@
+processing (as different rule types will be handled by specific engine modules).
+The signature types are defined in `src/detect.h
+<https://github.com/OISF/suricata/blob/master/src/detect.h>`_:
+
+.. literalinclude:: ../../../src/detect.h
+    :caption: src/detect.h
+    :language: c
+    :start-after: // rule types documentation tag start: SignatureType
+    :end-before: // rule types documentation tag end: SignatureType
+
+In more human readable terms:
+
+.. list-table:: Suricata Rule Types, and their Engine Analysis Term
+    :header-rows: 1
+
+    * - Rule Type
+      - Code Symbol
+      - Engine-Analysis Representation
+    * - Decoder Events Only
+      - ``SIG_TYPE_DEONLY``
+      - ``de_only``
+    * - Packet
+      - ``SIG_TYPE_PKT``
+      - ``pkt``
+    * - IP Only
+      - ``SIG_TYPE_IPONLY``
+      - ``ip_only``
+    * - IP Only (contains negated address(es))
+      - ``SIG_TYPE_LIKE_IPONLY``
+      - ``like_ip_only``
+    * -  Protocol Detection Only
+      - ``SIG_TYPE_PDONLY``
+      - ``pd_only``
+    * - Packet-Stream
+      - ``SIG_TYPE_PKT_STREAM``
+      - ``pkt_stream``
+    * - Stream
+      - ``SIG_TYPE_STREAM``
+      - ``stream``
+    * - Application Layer Protocol
+      - ``SIG_TYPE_APPLAYER``
+      - ``app_layer``
+    * - Application Layer Protocol Transactions
+      - ``SIG_TYPE_APP_TX``
+      - ``app_tx``
+
+The rule type will impact:
+
+  - To what does the signature action apply, in case of a match (`Action Scope`)
+  - When is the rule matched against traffic (`Inspected`)
+  - Against what the rule matches (`Matches`)
+
+This categorization is done taking into consideration the presence or absence of
+certain rule elements, as well as the type of keywords used. The categorization
+currently takes place in `src/detect-engine-build.c:void SignatureSetType()
+<https://github.com/OISF/suricata/blob/master/src/detect-engine-build.c#L1642-L1704>`_.
+
+The ``SignatureSetType()`` overall flow is described below:
+
+.. image:: intro/OverallAlgoHorizontal-20241127.png
+   :width: 600
+   :alt: A flowchart representing the SignatureSetType function.
+
+The following table lists all Suricata signature types, and how they impact the
+aspects aforementioned.
+
+.. list-table:: Suricata Rule Types
+    :widths: 10, 17, 18, 30, 25
+    :header-rows: 1
+
+    * - Type
+      - Action Scope
+      - Inspected
+      - Matches
+      - Keyword Examples (non-exhaustive)
+    * - Decoder Events Only
+      - Packet
+      - Per-packet basis
+      - Packets that are broken on an IP level
+      - ``decode-event``
+    * - Packet
+      - Packet
+      - Per-packet basis
+      - Packet-level info (e.g.: header info)
+      - ``itype``, ``tcp.hdr``, ``tcp.seq``, ``ttl`` etc.
+    * - IP Only
+      - Flow
+      - Once per direction
+      - On IP addresses on the flow
+      - Source/ Destination field of a rule
+    * - IP Only (contains negated address)(*)
+      - Flow
+      - Once per direction
```

Comment:
All packets.

Reply:
Test this with rule profiling to confirm.

Reply:
Maybe we can have a SV test with like-ip-only
- alert rule: several alerts
- drop/pass: should be applied to the flow